galley/pkg/config/analysis/msg/messages.yaml

```
@@ -396,3 +396,12 @@ messages:
     args:
       - name: gateways
         type: "[]string"
+
+  - name: "ServiceEntryHTTPSTrafficOnHTTPPort"
+    code: IST0139
+    level: Warning
+    description: "ServiceEntry Sending HTTPS to HTTP port."
+    template: "ServiceEntry Sending HTTPS to HTTP port on %s"
```

Would it make sense to report not just the host but also the port and DestinationRule that are in conflict?  Something like

`Using HTTPS for %s:%d conflicts with destination rule %s which specifies HTTP for port %d`
```suggestion
    template: "ServiceEntry Sending HTTPS to HTTP port on %s"
```
```suggestion
    template: "ServiceEntry Sending HTTPS to HTTP port on %s"
```

---

Hi @esnible not all ServiceEntry with related host and port have DestinationRule. That is why we cannot report the DestinationRule.

---

I was thinking about your second case, `serviceEntry.checkDestinationRulesPerformTLSOrigination()`.  If your code has identified the DestinationRule that is in conflict, it would be good to report that so the user doesn't struggle.

Consider either having two codes (with and without conflicting destination rule) or a message that includes an optional string.

`Using HTTPS for HTTP port %s:%d for HTTP: %s` where the final %s is set to either `"default"` or `"destination rule details.bookinfo"`

---

I am not sure about the DestinationRule conflict, can you leverage more? I am not sure probably my code is hard to read. but here is the logic:

1. loop for every host in ServiceEntry
2. get DestinationRule for the specific host
3. check ServiceEntry has protocol HTTP and port 443
4. if ServiceEntry has protocol HTTP and port 443, we need to check if there is DestinationRule that actually doing TLS origination (upgrade protocol HTTP to HTTP).
5. if there is no DestinationRule / DestinationRule that doesn't do TLS origination, we can return a warning message since it's forwarding HTTP over port 443.

---

I may have misread your code.  I spent the most time looking at your testdata and messages and less on analyzeProtocol() itself.  Some questions

- analyzeProtocol() only reports on port 443.  Should it also look at 8443 and other common HTTPS ports?
- Would it make sense to analyze a general case where the ServiceEntry `port.Protocol`, and port number doesn't match the results of your `checkDestinationRulesPerformTLSOrigination()`?

---

@esnible
1. Do you know which alternative port apart from 8443 for HTTPS?
2. the general cases is when we don't find the DestinationRule, you can check this one ```if len(relatedDestinationRules) == 0```